pkg/hvs/controllers/flavortemplate_controller.go

```diff
@@ -0,0 +1,327 @@
+		return createFlavorTemplateReq, &badRequestError{Message: "Unable to read request body"}
+	}
+
+	//Restore the request body to it's original state
+	r.Body = ioutil.NopCloser(bytes.NewBuffer(body))
+
+	//Decode the incoming json data to note struct
+	dec := json.NewDecoder(r.Body)
+	dec.DisallowUnknownFields()
+
+	err = dec.Decode(&createFlavorTemplateReq)
+	if err != nil {
+		defaultLog.WithError(err).Error("controllers/flavortemplate_controller:getFlavorTemplateCreateReq() Unable to decode request body")
+		return createFlavorTemplateReq, &badRequestError{Message: "Unable to decode request body"}
+	}
+
+	if createFlavorTemplateReq.ID != uuid.Nil {
+		template, err := ftc.Store.Retrieve(createFlavorTemplateReq.ID)
+		if err != nil {
+			if strings.Contains(err.Error(), commErr.RowsNotFound) {
+				defaultLog.Debug("controllers/flavortemplate_controller:getFlavorTemplateCreateReq() Flavor template with given ID is not exists,procceed with template creation")
+			} else {
+				defaultLog.WithError(err).Error("controllers/flavortemplate_controller:getFlavorTemplateCreateReq() Failed to retrieve flavor template")
+				return hvs.FlavorTemplate{}, errors.New("Failed to validate flavor template ID")
+			}
+		}
+		if template != nil || strings.Contains(err.Error(), "duplicate key value violates unique constraint") {
+			defaultLog.WithError(err).Error("controllers/flavortemplate_controller:getFlavorTemplateCreateReq() Unable to create flavor template, template with given template ID already exists or has been deleted")
+			return hvs.FlavorTemplate{}, &badRequestError{Message: "FlavorTemplate with given template ID already exists or has been deleted"}
+		}
+	}
+
+	defaultLog.Debug("Validating create flavor request")
+	errMsg, err := ftc.validateFlavorTemplateCreateRequest(createFlavorTemplateReq, string(body))
+	if err != nil {
+		return createFlavorTemplateReq, &badRequestError{Message: errMsg}
+	}
+
+	if len(createFlavorTemplateReq.Condition) == 0 {
+		defaultLog.WithError(err).Error("controllers/flavortemplate_controller:getFlavorTemplateCreateReq() Unable to create flavor template, empty condition field provided")
+		return hvs.FlavorTemplate{}, &badRequestError{Message: "Unable to create flavor template, empty condition field provided"}
+	}
+
+	return createFlavorTemplateReq, nil
+}
+
+// validateFlavorTemplateCreateRequest This method is used to validate the flavor template
+func (ftc *FlavorTemplateController) validateFlavorTemplateCreateRequest(FlvrTemp hvs.FlavorTemplate, template string) (string, error) {
+	defaultLog.Trace("controllers/flavortemplate_controller:validateFlavorTemplateCreateRequest() Entering")
+	defer defaultLog.Trace("controllers/flavortemplate_controller:validateFlavorTemplateCreateRequest() Leaving")
+	// Check whether the template is adhering to the schema
+	schemaLoader := gojsonschema.NewSchemaLoader()
+
+	definitionsSchemaJSON, err := readJSON(ftc.CommonDefinitionsSchema)
```

Comment:
Definitions and template need to be read at every create template operation, can we cache them?

Reply:
Schemas will be static and could be additional properties of the "ftc" (that are loaded once during the lifetime of HVS).